packages/wrangler/src/pages/functions/routing-rules-util.test.ts

```diff
@@ -0,0 +1,131 @@
+import {
+	isRoutingRuleMatch,
+	transformRoutingRuleToRegExp,
+} from "./routing-rules-util";
+
+describe("isRoutingRuleMatch", () => {
+	it("should match rules referencing root level correctly", () => {
+		const routingRule = "/";
+
+		expect(isRoutingRuleMatch("/", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("/foo/", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("/foo/bar", routingRule)).toBeFalsy();
+	});
+
+	it("should match include-all rules correctly", () => {
+		const routingRule = "/*";
+
+		expect(isRoutingRuleMatch("/", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/bar", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/bar/", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/bar/baz", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/bar/baz/", routingRule)).toBeTruthy();
+	});
+
+	it("should match `/*` suffix-ed rules correctly", () => {
+		let routingRule = "/foo/*";
+
+		expect(isRoutingRuleMatch("/foo", routingRule)).toBeTruthy();
```

Comment:
This surprises me. If it's like this on prod, then fair enough! But coming to this as a naive user, I wouldn't expect this to match. Same as L43.

Reply:
Checked our prod code, and yeah, that seems to be how we're handling it. Fair enough, and good job catching it.